### PR TITLE
Colorize search entities and filter buttons

### LINKED ISSUE #251: Consistent coloring of Nodes/Edges
UI part of #199.
Use color stores from #250.

### LINKED ISSUE #260: Add toggle or checkbox for colorizing graph
To fully implement #199 a checkbox or toggle to switch between grayscale and colorized graph has to be added

### LINKED ISSUE #261: Colorize filter buttons and enitities of the search
To fully implement #199, filter buttons and entities of the search need to be colorized the same way the graph is colored

### PR DESCRIPTION
**Pull request by derwehr**
This PR colorizes search entites and filter buttons and adds a toggler for entity colorization, closes #261, closes #260, closes #251

### CODE REVIEW

File: frontend/src/visualization/filtering/GreyScaleToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * Control that toggles colorization of the graph.
+ */
+export default function GreyScaleToggle(): JSX.Element {
+  const entityColorStore = useService(EntityColorStore);
+
+  // toggle greyScale
+  const [colorize, setGreyScale] = useState<boolean>(
+    !entityColorStore.getGreyScale()
+  );
```
Comment (joluj) [inline frontend/src/visualization/filtering/GreyScaleToggle.tsx:15]: use store + consitent naming
```suggestion
  const greyScale = useObservable(entityColorStore.getGreyScale());
```

later (lines 18-21) set it with:

```ts
   entityColorStore.setGreyScale(!greyScale);
```

File: frontend/src/stores/colors/EntityColorStore.ts [hunk 2]
```diff
@@ -142,6 +152,15 @@ export class EntityColorStore {
   public getValue(): EntityColorizer {
     return this.storeSubject.value;
   }
+
+  public getGreyScale(): boolean {
```
Comment (joluj) [inline frontend/src/stores/colors/EntityColorStore.ts:156]: add a sentence of documentation

**Review by joluj — CHANGES_REQUESTED** [review]
Store implementation missing. I have written the solution basically everywhere so just copy it:D only slight adaptations necessary.

Rest is very clean and easy to understand.

File: frontend/src/visualization/filtering/Filter.tsx [hunk 3]
```diff
@@ -85,6 +86,9 @@ const Filter = (): JSX.Element => {
 
   const filterStateStore = useService<FilterStateStore>(FilterStateStore);
 
+  const entityColorStore = useService(EntityColorStore);
+  const colorize = entityColorStore.getValue();
```
[the remark this answers is not in the document]
Comment (derwehr) [reply frontend/src/visualization/filtering/Filter.tsx:90]: Would that be the same as using the following?
```suggestion
  const colorize = useObservable(
    entityColorStore.getState(),
    entityColorStore.getValue()
  );
```
This makes sure colorize can't be undefined, else i get a ts error when invoking colorize later.

[the remark this answers is not in the document]
Comment (joluj) [reply frontend/src/visualization/filtering/Filter.tsx:90]: Yes, I forgot the initial value.

**Comment by derwehr** [comment]
Now this throws a bunch of runtime errors :cry: , will try to resolve them tomorrow

**Comment by joluj** [comment]
React/useState is behaves differently when called with an function. Maybe we should refactor EntityColorizer to be an object with a colourize function

**Review by joluj — APPROVED** [review]

**Pull request merged**